File: src/recipe_kitchen/schemas/jobs.py
from __future__ import annotations

from typing import Any, Literal

from pydantic import BaseModel

JobKind = Literal["ingest", "recipe", "audio", "video"]
JobState = Literal["queued", "running", "succeeded", "failed"]


class JobAccepted(BaseModel):
    job_id: str


class Job(BaseModel):
    id: str
    kind: JobKind
    status: JobState
    input: dict[str, Any] = {}
    result: dict[str, Any] | None = None
    error: str = ""


class JobStatus(BaseModel):
    job_id: str
    kind: JobKind
    status: JobState
    result: dict[str, Any] | None = None
    error: str = ""

File: frontend/lib/types/jobs.ts
export type JobKind = "ingest" | "recipe" | "audio" | "video";
export type JobState = "queued" | "running" | "succeeded" | "failed";

export type JobAccepted = {
  job_id: string;
};

export type JobStatus<T = unknown> = {
  job_id: string;
  kind: JobKind;
  status: JobState;
  result: T | null;
  error: string;
};
